### DISCUSSION TITLE
Is there any way to show local html file？

### DISCUSSION THREAD

**Discussion by Nash-x9** [tauri-apps/tauri#8733] [Q&A]
thks for your awesome project
I want to use tauri to create a desktop program, that support display local html file.
I had try to use iframe,set iframe src like `$HOME/.tarui_app/third.html`,it can't load..

some advice please...THKS AGAIN.

**Comment by FabianLars** [comment]
hmm, maybe try setting the asset protocol as the iframe source via https://tauri.app/v1/api/js/tauri/#convertfilesrc

**Reply by Nash-x9** [reply]
thks bro
there's some trouble.
1. when I use iframe, app will warn`[TAURI] Couldn't find callback id 3551468386 in window. This happens when the app is reloaded while Rust is running an asynchronous operation.`
2. iframe doesn't support `assets` protocol,error message `Failed to load resource: unsupported URL`

btw, It will autocreate appDataDir when I start App?

**Reply by HuakunShen** [reply]
@FabianLars I tried `convertFileSrc`, it could load the html, but not the `.js` and `.css` files.

`convertFileSrc` return absolute path like `asset://localhost//<paths>/index.html`. In the `index.html` dependencies are loaded with relative path
```html
<script type="module" crossorigin src="./assets/index-CZGpdclX.js"></script>
<link rel="stylesheet" crossorigin href="./assets/index-BHYCjRZw.css">
```

But the webview tries to load `asset://localhost/assets/index-BHYCjRZw.css` and `asset://localhost/assets/index-CZGpdclX.js`

The `<path>` section is omited. Do you know any way to work around this?

Since the absolute path on each computer is different, I can't simply change the base URL for the loaded web page.

**Comment by ilyazub** [comment]
It's possible with `tauri-apps/wry`: https://github.com/tauri-apps/wry/discussions/761#discussioncomment-7579658. (_It's the underlying cross-platform WebView library_.)

**Comment by Salman-Sali** [comment]
Hey @Nash-x9 
Have you solved this? How did you solve?
